protocol/src/debugProtocol.ts

```diff
@@ -411,6 +411,26 @@ export module DebugProtocol {
 	export interface RestartFrameResponse extends Response {
 	}
 
+	/** Goto request; value of command field is "goto".
+		The request sets the location where the debuggee will continue to run.
+		This makes it possible to skip the execution of code or to executed code again.
+		The code between the current location and the goto target is not executed but skipped.
+		The debug adapter first sends the GotoResponse and then a StoppedEvent (event type 'goto').
+	*/
+	export interface GotoRequest extends Request {
+		arguments: GotoArguments;
+	}
+	/** Arguments for "goto" request. */
+	export interface GotoArguments {
+		/** Set the goto target for this thread. */
+		threadId: number;
+		/** The location where the debuggee will continue to run. */
+		targetId: number;
+	}
+	/** Response to "goto" request. This is just an acknowledgement, so no body field is required. */
+	export interface GotoResponse extends Response {
+	}
```

Comment:
The semantics of this protocol are a bit different than the way VS works. In VS we don't retrieve the list of valid targets first because it could potentially be very large (some customers have large functions, especially in auto generated code. So instead, the ui attempts the set next statement wherever the user has the cursor and then fails if it is an invalid location. We may want to consider that model since this could be a very large response.